data/moves.js

```diff
@@ -9627,15 +9627,15 @@ let BattleMovedex = {
 			onStart: function (target, source, effect) {
 				this.add('-singleturn', target, 'move: Magic Coat');
 				if (effect && effect.effectType === 'Move') {
-					this.effectData.pranksterBoosted = effect.pranksterBoosted;
+					this.effectData.pranksterBoosted = /** @type {ActiveMove} */ (effect).pranksterBoosted;
```

Comment:
Out of interest, is there a more Typescript-safe way to do this check (I'm thinking `effect instanceof ActiveMove`, if it could be made to work)?

Reply:
If we can assure that an `Effect`'s `Move` is always `Active`, we would just update it there. Can we assure that?

Reply:
I think we can, in practice.